### PR TITLE
Make Solidity scripts more betterer

### PR DESCRIPTION
**Pull request by drmercer**
**Don't merge until #18 is merged (this branch includes those changes).** Also don't merge this branch into `solidity` until that PR is merged, so that these changes don't get mixed in with the other changes in that PR.

Added compilation error/warning reporting to compile script.
Improved the compile script so that it only recompiles if needed.
Added a `solidity:clean` script.
Added the beginnings of unit tests for the frontend solidity API.

### CODE REVIEW

**Comment by joshuakcockrell** [comment]
@drmercer Do you want me to merge this now? I merged the other PR.

File: package.json [hunk 1]
```diff
@@ -65,13 +65,14 @@
   },
   "scripts": {
     "build": "node scripts/build.js",
-    "solidity:compile": "node solidity/compile.js",
+    "solidity:clean": "node solidity/scripts/clean.js",
```
Comment (smithskyler) [inline package.json]: I like how you moved the solidity scripts to a separate directory. It's a step in the right direction for making the solidity section an independent module.

File: solidity/facades/frontend.js [hunk 2]
```diff
@@ -6,9 +6,9 @@
 const address = require('../deployed_main_contract.json').address;
 const abi = require('./build_abis/CryptoShapeMain_abi.json');
 
-module.exports = function(web3) {
+module.exports = function(web3, addressOverride) {
 
-	const mainContract = new web3.eth.Contract(abi, address);
+	const mainContract = new web3.eth.Contract(abi, addressOverride || address);
```
Comment (smithskyler) [inline solidity/facades/frontend.js]: I was unfamiliar with this || functionality is JS. That's cool.

File: solidity/scripts/compile.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+	.map(f => path.resolve(contractsPath, f))
+	.map(p => fs.statSync(p).mtime > lastBuildTime)
+	.reduce((a, b) => a || b);
+if (hasChanges) {
+	removeBuildOutput();
+} else {
+	console.log(`No changes made to contracts since last build`);
+	return; // Stop executing
+}
+
```
Comment (smithskyler) [inline solidity/scripts/compile.js]: This is a cool function you made here to automatically compile the contract only when it needs to be recompiled. I'll want you to explain the functionality to me.

File: solidity/tests/frontend.test.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+describe("Frontend Solidity API", () => {
+	describe("getShapeCount()", () => {
+		it("should return the number of shapes in the blockchain", async () => {
+			const count = await frontendAPI.getShapeCount();
+			assert.equal(count, 2);
+		});
+	});
+
+	describe("personalSign(...)", () => {
```
Comment (smithskyler) [inline solidity/tests/frontend.test.js]: What is personalSign for?

**Comment by drmercer** [comment]
FYI: I believe these changes introduced a build-breaking bug, because I didn't update the paths in the deploy script. I fixed that bug in 3d0e901b1f7c7b664d1bc6bab0cfccde5dea3674.

**Pull request merged**